js/data/bucket/symbol_bucket.js

```diff
@@ -339,23 +339,20 @@ SymbolBucket.prototype.placeFeatures = function(collisionTile, showCollisionBoxe
 
     this.createArrays();
 
-    var elementGroups = this.elementGroups = {
-        glyph: [],
-        icon: [],
+    var layout = this.layer.layout;
+
+    this.metadata = {
         sdfIcons: this.sdfIcons,
-        iconsNeedLinear: this.iconsNeedLinear
+        iconsNeedLinear: this.iconsNeedLinear,
+        adjustedTextSize: this.adjustedTextSize,
+        adjustedIconSize: this.adjustedIconSize,
+        // Transfer the name of the fonstack back to the main thread along with the buffers.
+        // The draw function needs to know which fonstack's glyph atlas to bind when rendering.
+        fontstack: layout['text-font'].join(',')
```

Comment:
These are the properties that need to be serialized. Putting them in an objects makes it unnecessary to serialize/deserialize them individually. This is good because the serialization code is in Bucket and these individual properties are specific to SymbolBucket